src/LinkedListVisualizer/LinkedListVisualizer.jsx

```diff
@@ -0,0 +1,497 @@
+                currWalkNode.className = 'node-walk'; 
+                prevWalkNode = currWalkNode;
+                walkCount++;
+            } else {
+                currWalkNode = document.getElementById(`node-${i % numberOfNodes}`);
+                currWalkNode.className = 'node-walk'; 
+                prevWalkNode = currWalkNode;
+            }                
+            i++;
+
+            if(i !== 0 && i  !== 1 && currWalkNode == currRunNode){
+                prevWalkNode.className = 'node';
+                prevRunNode.className = 'node';
+                currWalkNode.className = 'node-join';
+                clearInterval(timer); 
+                return;
+            }
+
+
+        }
+
+        checker();
+        timer = setInterval(checker, 1000);
+    }
+
+    handleNodeDataChange = idx => event => {
+        event.preventDefault();
+        const newNodeData = this.state.inputGrid.map((inputBox, bidx) => {
+            if(idx !== bidx) return inputBox;
+            return {...inputBox, data: event.target.value};
+        })
+
+        this.setState({ inputGrid : newNodeData});
+    }
+
+    handleAddNode = (event) => {
+        event.preventDefault();
+        const {numberOfNodes} = this.state;
+        const newData = this.state.inputGrid.pop().data;
+        const nodeLimit = (this.state.windowWidth - 150) / 150;
+        
+        if(newData > 999 || newData < 0){
+            alert("This is out of bounds. Please enter a value between 0 and 999.");
+        } else if(!this.state.cyclePresent && numberOfNodes < nodeLimit){
+            this.setState({
+                grid : this.state.grid.concat([createNode(newData, this.state.numberOfNodes)]),
+                numberOfNodes : this.state.numberOfNodes + 1
+            }, () => {console.log(this.state.numberOfNodes, 'numberOfNodes')})    
+
+        } else if(numberOfNodes >= nodeLimit) {
+            alert("Cannot add node, exceeds node limit for browser window size.")
+        } else{
+            alert("Cannot add node, cycle is present.");
+        }
+        this.setState({inputGrid : this.state.inputGrid.concat([{ data: "" }])})
+    }
+
+    handleCycleBox = (event) => {
+        event.preventDefault();
+        this.setState({
+            cycleToHolder : event.target.value
+        })    
+    }
+
+    handleAddCycle = (event) => {
+        event.preventDefault();
+        const {cycleTo, grid, cycleToHolder, numberOfNodes} = this.state;
+        if(this.visualizationInProcess()){
+            alert("Please wait for current visualization to end before proceeding.");
+        } else if(numberOfNodes > 1 && cycleToHolder >= 0 && cycleToHolder < numberOfNodes - 1){
+            const elementNode = document.getElementById('node-0');
+            const styleNode = window.getComputedStyle(elementNode);
+            const widthNode = parseInt(styleNode.getPropertyValue('width').replace('px',''));
```

Comment:
This data is flowing in the opposite direction; your component should decide what width the node *should* have and use that to style the node (probably based on the window dimensions). Here you are relying on what width the node gets assigned by the browser and using that to inform your component.